ValveResourceFormat/Resource/ResourceTypes/Choreo/Parser/BVCDParser.cs

```diff
@@ -0,0 +1,359 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ValveResourceFormat.ResourceTypes.Choreo.Enums;
+using ValveResourceFormat.ResourceTypes.Choreo.Flags;
+using ValveResourceFormat.ResourceTypes.Choreo.Data;
+
+namespace ValveResourceFormat.ResourceTypes.Choreo.Parser
+{
+    public abstract class BVCDParser
+    {
+        public abstract byte Version { get; }
+        private BinaryReader reader;
+        private string[] strings;
+
+        protected BVCDParser(BinaryReader reader, string[] strings)
+        {
+            this.reader = reader;
+            this.strings = strings;
+        }
+
+        public static ChoreoData Parse(Stream stream, string[] strings)
+        {
+            using BinaryReader reader = new BinaryReader(stream);
+
+            var magic = new string(reader.ReadChars(4));
+            if (magic != "bvcd")
+            {
+                throw new InvalidDataException("The content of the given stream is not bvcd data");
+            }
+            var version = reader.ReadByte();
+            var crc = reader.ReadInt32();
+
+            BVCDParser parser;
+            switch (version)
+            {
+                case 8: //There are some v8 bvcds in HLA, but they decompile fine with the v9 parser.
+                case 9:
+                    parser = new BVCDParserV9(reader, strings);
```

Comment:
I'd rather not create new classes for each version. Usually version bumps just add/modify fields and its one `if` away.